spec.bs

```diff
@@ -557,7 +574,31 @@ To <dfn>append private state token redemption request headers</dfn> given a [=/r
 
  1. [=Set redemption headers=] with |request| and |record|.
 
-ISSUE(165): Add redemption response handling, including calling [=record redemption timestamp=].
+
+Handling Redeem Responses {#redeem-response}
+----------------------------------------------------------
+
+To <dfn>handle a redeem response</dfn>, given [=request=] |request| and [=response=] |response|, run the following steps:
+
+1. If |request|'s [=request/header list=] [=header list/contains|does not contain=] <a http-header>Sec-Private-State-Token</a>, return null.
+1. If |response|'s [=response/header list=] [=header list/contains|does not contain=] <a http-header>Sec-Private-State-Token</a>, return a [=network error=].
+1. Let |header| be the result of [=header list/get|getting=] <a http-header>Sec-Private-State-Token</a> from |response|'s [=response/header list=].
+1. If |header| is empty, return null.
+1. [=header list/delete|Delete=] <a http-header>Sec-Private-State-Token</a> from |response|'s [=response/header list=].
+1. Set |lifetime| to be the largest representable duration.
+1. If |response|'s [=response/header list=] [=header list/contains=] <a http-header>Sec-Private-State-Token-Lifetime</a> response header, set |lifetime| to that value.
+1. [=header list/delete|Delete=] <a http-header>Sec-Private-State-Token-Lifetime</a> from |response|'s [=response/header list=].
+1. Let |issuer| be |request|'s [=request/URL=]'s [=url/origin=].
+1. Let |topLevel| be |request|'s [=request/client=]'s [=environment/top-level origin=].
+1. Perform [=record redemption timestamp=] with |issuer| and |topLevel|. 
+1. Perform [=record a redemption record=] with |issuer|, |topLevel|, |header|, and |lifetime|.
+
+<h3 id="modification-to-http-fetch-steps-header-redeem">Modification to HTTP fetch steps</h3>
+The following steps will be added to the [=HTTP fetch=] steps, before checking the redirect status (i.e. "7. If |actualResponse|'s status is a redirect status, ..."):
+
+1. Let |redeem response result| be the result of [=handle a redeem response|handling a redeem response=], given [=request=] |request| and [=response=] |actualResponse| as input.
+1. If |redeem response result| is a [=network error=], return |issue response result|.
+
 
 <a http-header>Sec-Private-State-Token-Lifetime</a> response header indicates how long (in seconds) the
     [=redemption record=] should be cached for. When <a http-header>Sec-Private-State-Token-Lifetime</a> response header value
```

Comment:
Now that we have an algorithm, can we delete this note here (the entire paragraph) or convert parts of it to a `Note:`? Before doing that, it makes sense to double-check that we're including everything that's needed in the algorithm above, e.g.

```
If
<a http-header>Sec-Private-State-Token-Lifetime</a> header is omitted, the lifetime of the [=redemption record=] will be
tied to the lifetime of the Private State Token verification key that confirmed the
redeemed token's issuance.
```

seems inconsistent now?

Reply:
Missed the key expiration part, added it back.